test/it/uk/gov/hmrc/exports/mongock/changesets/CacheChangeLogSpec.scala

```diff
@@ -34,9 +37,13 @@ class CacheChangeLogSpec extends WordSpec with MustMatchers with GuiceOneServerP
 
   private val changeLog = new CacheChangeLog()
 
+  private val idGenerator = mock[IdGenerator[String]]
+  changeLog.setIdGenerator(idGenerator)
```

Comment:
It's used here @PatrykRudnicki - some sort of code injection. :-)

Reply:
CacheChangeLog is a normal scala class. I think we can use it in a way like always `class CacheChangeLog @Inject()(idGenerator: IdGenerator)`.

Additionally creating a methods just for tests are probably not a good idea.

In general it's inside the cache changes/migrations, so it's not a big deal, so it's non blocking comment

Reply:
You cannot use any Injection framework in ChangeLogs. Unfortunatelly, they are dependent on `Mongock` object and it is the expected behavior that ChangeLogs are indpendent from any framework.
I would agree with adding a method just for test, but I don't see much options here. Besides this class is not widely used, it's scope is limited so it should not do much harm to have it as it is.

Reply:
```
private val defaultGenerator: IdGenerator[String] = new StringIdGenerator
  private def idGenerator(generator: IdGenerator[String] = defaultGenerator): IdGenerator[String] = generator
```

In that case probably enough to use this approach

Reply:
but this won't give a possibility to use it in tests. Okay, out of ideas right now